### PR TITLE
Update to Minecraft v1.19.10 (Protocol v534)

### PR DESCRIPTION
**Pull request by GoodLucky777**
Update to Minecraft v1.19.10 (Protocol v534) (#1385)

- Update resources
- Update protocol related things (v534)
- Update `BlockLapis.getName()`
- Don't show unknown recipe warning for not implemented things

### CODE REVIEW

File: src/main/java/cn/nukkit/network/protocol/LevelSoundEventPacket.java [hunk 1]
```diff
@@ -228,12 +231,14 @@ public class LevelSoundEventPacket extends DataPacket {
     public static final int SOUND_HURT_BABY = 219;
     public static final int SOUND_DEATH_BABY = 220;
     public static final int SOUND_STEP_BABY = 221;
-    @PowerNukkitOnly @Since("1.6.0.0-PN") public static final int SOUND_BABY_SPAWN = 222;
+    public static final int SOUND_SPAWN_BABY = 222;
+    @Deprecated @PowerNukkitOnly @Since("1.6.0.0-PN") public static final int SOUND_BABY_SPAWN = SOUND_SPAWN_BABY;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/LevelSoundEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/LevelSoundEventPacket.java [hunk 2]
```diff
@@ -228,12 +231,14 @@ public class LevelSoundEventPacket extends DataPacket {
...
     public static final int SOUND_BORN = 223;
     public static final int SOUND_BLOCK_TURTLE_EGG_BREAK = 224;
     public static final int SOUND_BLOCK_TURTLE_EGG_CRACK = 225;
     public static final int SOUND_BLOCK_TURTLE_EGG_HATCH = 226;
-    @PowerNukkitOnly @Since("1.6.0.0-PN") public static final int SOUND_TURTLE_LAY_EGG = 227;
+    public static final int SOUND_LAY_EGG = 227;
+    @Deprecated @PowerNukkitOnly @Since("1.6.0.0-PN") public static final int SOUND_TURTLE_LAY_EGG = SOUND_LAY_EGG;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/LevelSoundEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 3]
```diff
@@ -59,19 +64,27 @@ public class EntityEventPacket extends DataPacket {
...
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int SUMMON_AGENT = 73;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int FINISHED_CHARGING_CROSSBOW = 74;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int LANDED_ON_GROUND = 75;
-    
+    public static final int START_SWIM = 70;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int START_SWIMMING = START_SWIM;
+    public static final int BALLOON_POP = 71;
+    public static final int TREASURE_HUNT = 72;
+    public static final int AGENT_SUMMON = 73;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int SUMMON_AGENT = AGENT_SUMMON;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 4]
```diff
@@ -59,19 +64,27 @@ public class EntityEventPacket extends DataPacket {
...
+    public static final int CHARGED_CROSSBOW = 74;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int FINISHED_CHARGING_CROSSBOW = CHARGED_CROSSBOW;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 5]
```diff
@@ -59,19 +64,27 @@ public class EntityEventPacket extends DataPacket {
...
+    public static final int FALL = 75;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int LANDED_ON_GROUND = FALL;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 6]
```diff
@@ -59,19 +64,27 @@ public class EntityEventPacket extends DataPacket {
...
     public static final int MERGE_ITEMS = 69;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int START_SWIMMING = 70;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int BALLOON_POP = 71;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int TREASURE_HUNT = 72;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int SUMMON_AGENT = 73;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int FINISHED_CHARGING_CROSSBOW = 74;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int LANDED_ON_GROUND = 75;
-    
+    public static final int START_SWIM = 70;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int START_SWIMMING = START_SWIM;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 7]
```diff
@@ -59,19 +64,27 @@ public class EntityEventPacket extends DataPacket {
     public static final int DEATH_SMOKE_CLOUD = 61;
     public static final int COMPLETE_TRADE = 62;
     public static final int REMOVE_LEASH = 63;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int CARAVAN = 64;
+    public static final int CARAVAN_UPDATED = 64;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int CARAVAN = CARAVAN_UPDATED;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 8]
```diff
@@ -34,18 +35,22 @@ public class EntityEventPacket extends DataPacket {
...
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int GUARDIAN_ATTACK_ANIMATION = GUARDIAN_ATTACK;
     public static final int WITCH_DRINK_POTION = 29;
     public static final int WITCH_THROW_POTION = 30;
     public static final int MINECART_TNT_PRIME_FUSE = 31;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int PRIME_CREEPER = 32;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int AIR_SUPPLY = 33;
+    public static final int CREEPER_PRIME_FUSE = 32;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int PRIME_CREEPER = CREEPER_PRIME_FUSE;
+    public static final int AIR_SUPPLY_EXPIRED = 33;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int AIR_SUPPLY = AIR_SUPPLY_EXPIRED;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 9]
```diff
@@ -34,18 +35,22 @@ public class EntityEventPacket extends DataPacket {
...
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int GUARDIAN_ATTACK_ANIMATION = 28;
+    public static final int GUARDIAN_ATTACK = 28;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int GUARDIAN_ATTACK_ANIMATION = GUARDIAN_ATTACK;
     public static final int WITCH_DRINK_POTION = 29;
     public static final int WITCH_THROW_POTION = 30;
     public static final int MINECART_TNT_PRIME_FUSE = 31;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int PRIME_CREEPER = 32;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int AIR_SUPPLY = 33;
+    public static final int CREEPER_PRIME_FUSE = 32;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int PRIME_CREEPER = CREEPER_PRIME_FUSE;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 10]
```diff
@@ -34,18 +35,22 @@ public class EntityEventPacket extends DataPacket {
...
+    public static final int VILLAGER_HAPPY = 23;
     public static final int WITCH_SPELL_PARTICLES = 24;
     public static final int FIREWORK_EXPLOSION = 25;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int IN_LOVE_HEARTS = 26;
+    public static final int IN_LOVE_PARTICLES = 26;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int IN_LOVE_HEARTS = IN_LOVE_PARTICLES;
     public static final int SILVERFISH_SPAWN_ANIMATION = 27;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int GUARDIAN_ATTACK_ANIMATION = 28;
+    public static final int GUARDIAN_ATTACK = 28;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int GUARDIAN_ATTACK_ANIMATION = GUARDIAN_ATTACK;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 11]
```diff
@@ -34,18 +35,22 @@ public class EntityEventPacket extends DataPacket {
...
     public static final int LOVE_PARTICLES = 21;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int VILLAGER_ANGRY = 22;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int VILLAGER_HAPPY = 23;
+    public static final int VILLAGER_ANGRY = 22;
+    public static final int VILLAGER_HAPPY = 23;
     public static final int WITCH_SPELL_PARTICLES = 24;
     public static final int FIREWORK_EXPLOSION = 25;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int IN_LOVE_HEARTS = 26;
+    public static final int IN_LOVE_PARTICLES = 26;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int IN_LOVE_HEARTS = IN_LOVE_PARTICLES;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

File: src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java [hunk 12]
```diff
@@ -11,13 +11,14 @@
...
+
     @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int NONE = 0;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int JUMP = 1;
+    public static final int JUMP = 1;
     public static final int HURT_ANIMATION = 2;
     public static final int DEATH_ANIMATION = 3;
     public static final int ARM_SWING = 4;
-    @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int ATTACK_STOP = 5;
+    public static final int STOP_ATTACK = 5;
+    @Deprecated @PowerNukkitOnly @Since("1.4.0.0-PN") public static final int ATTACK_STOP = STOP_ATTACK;
```
Comment (LoboMetalurgico) [inline src/main/java/cn/nukkit/network/protocol/EntityEventPacket.java]: Please add @DeprecationDetails.

**Review by LoboMetalurgico — CHANGES_REQUESTED** [review]
Hello, thanks for your Pull Request.
Thanks again for the update, but before accepting, I would like you to review the new @Deprecated and add @DeprecationDetails.

**Comment by GoodLucky777** [comment]
Ready to be reviewed

**Review by LoboMetalurgico — APPROVED** [review]
Thank you!

**Pull request merged**